internal/orchestrator/orchestrator.go

```diff
@@ -452,46 +456,88 @@ func (or *orchestrator) initPlugins(ctx context.Context) (err error) {
 	}
 
 	if or.tokens == nil {
-		or.tokens = make(map[string]tokens.Plugin)
-		tokensConfigArraySize := tokensConfig.ArraySize()
-		for i := 0; i < tokensConfigArraySize; i++ {
-			prefix := tokensConfig.ArrayEntry(i)
-			name := prefix.GetString(tokens.TokensConfigName)
-			pluginName := prefix.GetString(tokens.TokensConfigPlugin)
-			if name == "" {
-				return i18n.NewError(ctx, coremsgs.MsgMissingTokensPluginConfig)
-			}
-			if err = core.ValidateFFNameField(ctx, name, "name"); err != nil {
-				return err
-			}
+		if err = or.initTokens(ctx); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (or *orchestrator) initTokens(ctx context.Context) (err error) {
+	or.tokens = make(map[string]tokens.Plugin)
+	tokensConfigArraySize := tokensConfig.ArraySize()
+	for i := 0; i < tokensConfigArraySize; i++ {
+		prefix := tokensConfig.ArrayEntry(i)
+		name := prefix.GetString(tokens.TokensConfigName)
+		pluginType := prefix.GetString(tokens.TokensConfigType)
+		if name == "" || pluginType == "" {
+			return i18n.NewError(ctx, coremsgs.MsgMissingTokensConfig)
+		}
+
+		if err = core.ValidateFFNameField(ctx, name, "name"); err != nil {
+			return err
+		}
+
+		log.L(ctx).Infof("Loading tokens plugin name=%s type=%s", name, pluginType)
+		pluginPrefix := prefix.SubPrefix(pluginType)
+
+		plugin, err := tifactory.GetPlugin(ctx, pluginType)
+		if plugin != nil {
+			err = plugin.Init(ctx, name, pluginPrefix, &or.bc)
+		}
+		if err != nil {
+			return err
+		}
+		or.tokens[name] = plugin
+	}
+
+	if len(or.tokens) > 0 {
+		return nil
+	}
+
+	// If there still is no tokens config, check the deprecated structure for config
```

Comment:
So we only parse the new config _or_ the old config (we don't interpolate them). That seems reasonable, just need to make sure it's called out in the release notes.